ADR/nnnn-revised-component-model.md

```diff
@@ -0,0 +1,59 @@
+# NN. Revised Component Model
+
+## Status
+
+Proposed
+[//]: # (One of: Proposed, Withdrawn, Deferred, Accepted, Implementable, Implemented, Replaced)
+
+## Context
+
+The existing application and component model in Konflux presents several
+limitations that hinder our ability to scale and support flexible workflows.
+Specifically, the current model requires a new component to be registered for
+each use case, even when the same underlying code is being used in multiple
+release scenarios. This creates unnecessary overhead and complexity, while also
+limiting the reusability of our components.
+
+Furthermore, ownership of the current APIs is unclear. They were originally
+owned by HAS (which is now deprecated) and exhibit excessive coupling across
+multiple services.
+
+## Decision
+
+We have decided to adopt a revised component model that replaces the concept of
+an "application" with a "component group." The key elements of this new model
+are:
+
+* **Component Branch**: A component can have one or more component branches,
```

Comment:
When was the idea of multi-branch components discussed? This was a key thing in all proposals IIRC.

Reply:
Maybe I'm misunderstanding, but "Component Branch" seems to be replacing the existing components, and the names seems to indicate its pointing at a particular branch. That would mean we would need to create a resource for each branch like we do today. I thought the point of enable multi-branch components was to not have to create new resources when branching.